powertools-tracing/src/main/java/software/amazon/lambda/powertools/tracing/TracingUtils.java

```diff
@@ -36,6 +39,7 @@ public final class TracingUtils {
      * @param value the value of the annotation
      */
     public static void putAnnotation(String key, String value) {
+        validateAnnotationKey(key);
```

Comment:
Should `validateAnnotationKey` returns `true`/`false`, and skip calling `putAnnotation` if it is false? 

From the issue description it seems like the behavior will be the same, but it would make it more explicit when reading through what's going on.

Reply:
+1

Reply:
I originally did this but changed it because I needed to have if branching and log statements in two methods rather than consolidating in one place.  But I've changed it back now.